Assets/MRTK/SDK/Features/UX/Scripts/InteractiveElement/BaseInteractiveElement.cs

```diff
@@ -230,5 +286,38 @@ public void SetEventConfigurationInstance(string stateName)
                 Debug.LogError($"{stateName} is not contianted in the States list");
             }
         }
+
+        /// <summary>
+        /// Add a Near Interaction Touchable component to the current game object if the Touch state is 
+        /// added to the States list. A Near Interaction Touchable component is required for an object to detect
+        /// touch input events. 
+        /// A Near Interaction Touchable Volume component is attached by default because it detects touch input
+        /// on the entire surface area of a collider.  While a Near Interaction Touchable component
+        /// will be attached if the object is a button because touch input is only detected within the area of a plane. 
+        /// </summary>
+        public void AddNearInteractionTouchable()
+        {
+            if (gameObject.GetComponent<BaseNearInteractionTouchable>() == null)
+            {
+                // Add a Near Interaction Touchable Volume by default because it detects touch on the 
+                // entire surface area of a collider. 
+                gameObject.AddComponent<NearInteractionTouchableVolume>();
+
+                // Add a Near Interaction Touchable if the object is a button.
+                // A Near Interaction Touchable detects touch input within the area of a plane and not the 
+                // entire surface area of an object.
+            }
```

Comment:
Once the button class is added, there will be a condition determining whether to add a Near Interaction Touchable Volume or a Near Interaction Touchable component.